### PR TITLE
Fix missing null check when removing lifecycle observer

### PR DESCRIPTION
**Pull request by markushi**
## :scroll: Description
If `Sentry.close` is called off the main thread, the lifecycle watcher field is set to `null`, while de-registering is posted on the main thread, leading to a NPE in combination with some versions of android x lifecycle lib.

## :bulb: Motivation and Context
Fixes https://github.com/getsentry/sentry-java/issues/2601

## :green_heart: How did you test it?
Manual testing, the overhead of running a single test against different version of the androidx lifecycle library didn't seem to justify the efforts.

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java [hunk 1]
```diff
@@ -110,23 +110,27 @@ private void addObserver(final @NotNull IHub hub) {
   }
 
   private void removeObserver() {
-    ProcessLifecycleOwner.get().getLifecycle().removeObserver(watcher);
+    final @Nullable LifecycleWatcher watcherRef = watcher;
```
Comment (stefanosiano) [inline sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java:113]: Is the `watcherRef` necessary? An `if` on the `watcher` should be enough, shouldn't it?

**Review by stefanosiano — COMMENTED** [review]
Can we also add a test to verify the change? like verifying the watcher is not null after the `close` function is called in a background thread and it's null only after the main thread executes?

File: sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java (hunk 1 above)
Comment (markushi) [reply sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java:113]: Yes, as watcher could be set to `null` on another thread

File: sentry-android-core/src/test/java/io/sentry/android/core/AppLifecycleIntegrationTest.kt [hunk 2]
```diff
@@ -94,4 +96,23 @@ class AppLifecycleIntegrationTest {
...
+
+        assertNotNull(sut.watcher)
+
+        Thread {
+            sut.close()
+            latch.countDown()
+        }.start()
+
+        latch.await()
+        InstrumentationRegistry.getInstrumentation().waitForIdleSync()
```
Comment (stefanosiano) [inline sentry-android-core/src/test/java/io/sentry/android/core/AppLifecycleIntegrationTest.kt:115]: For the main looper it should be possible to use `shadowOf(Looper.getMainLooper()).idle()` instead of creating a thread manually. It would be easier for readability if it works

**Review by stefanosiano — APPROVED** [review]
just a minor comment, if possible.
Looks good for everything else!
